arch/arm/src/imxrt/Kconfig

```diff
@@ -151,6 +151,47 @@ config ARCH_FAMILY_IMXRT106x
 	select ARMV7M_HAVE_DTCM
 	select IMXRT_HIGHSPEED_GPIO
 
+config IMXRT_HAVE_OTA_PARTITION
+	bool
+	default n
+
+config IMXRT_PROGMEM
+	bool "Flash progmem support"
+	default n
+	select ARCH_HAVE_PROGMEM
+	---help---
+		Add progmem support, start block and end block options are provided to
+		obtain an uniform flash memory mapping.
+
+menu "Application Image Configuration"
+choice
+	prompt "Application Image Format"
+	default IMXRT_APP_FORMAT_LEGACY
+	---help---
+		Depending on the chosen 2nd stage bootloader, the application may
+		be required to be perform a specific startup routine. Furthermore,
+		the image binary must be formatted according to the definition from
+		the 2nd stage bootloader.
```

Comment:
Indeed, if the chip does not really require a 2nd stage bootloader, then this `choice` may be removed and the `config IMXRT_APP_FORMAT_MCUBOOT` could be extracted from it.

Reply:
@xiaoxiang781216 why did you mark it as resolved? :-)

Reply:
I was looking into this  XXX_APP_FORMAT_LEGACY and was wondering why it is there. STM32 seems to have it, but not using it. Can I remove it from Kconfig?

Reply:
Actually Application Image Format choice has influence on flash linker script, where IMXRT_APP_FORMAT_MCUBOOT instructs linker to make changes in the linking process for the loader and app respectively. In case IMXRT_APP_FORMAT_MCUBOOT  is missing, thus IMXRT_APP_FORMAT_LEGACY is choosen, linker calls ordinary flash script for linking. I would say IMXRT_APP_FORMAT_LEGACY should be included, although does not have explicit usage.

Reply:
> Does iMX-RT has a bootloader to be flashed before the application? I'm not sure. I think probably STM32 and iMX-RT are getting a definition that was used for ESP32 and replicating without needing it. What do you think @gustavonihei ?

The mcuboot-loader configuration builds a bootloader, which is then flashed to address 0x70000000 as a second stage bootloader. First stage bootloader is in ROM. The mcuboot-app configuration is used to build application image, linked to 0x70040000 address. This image is verified and loaded by second stage bootloader. So yes, loader does need to be flashed before application image.

Reply:
> Actually Application Image Format choice has influence on flash linker script, where IMXRT_APP_FORMAT_MCUBOOT instructs linker to make changes in the linking process for the loader and app respectively. In case IMXRT_APP_FORMAT_MCUBOOT is missing, thus IMXRT_APP_FORMAT_LEGACY is choosen, linker calls ordinary flash script for linking. I would say IMXRT_APP_FORMAT_LEGACY should be included, although does not have explicit usage.

`IMXRT_APP_FORMAT_LEGACY` is not required to achieve what you described. With the modification I suggested in https://github.com/apache/nuttx/pull/8684#discussion_r1121697547 it is also possible to select the correct linker script with a simple test for `IMXRT_APP_FORMAT_MCUBOOT`.

Like @acassis said, leaving `IMXRT_APP_FORMAT_LEGACY` option when a 2nd stage bootloader is not present is  incoherent.

Reply:
Ok, I just noticed that you've edited the help text and removed the reference to a 2nd stage bootloader.
It is fine to leave the `choice` with both options.